ExampleMod/Common/ExamplePlayerDrawLayer.cs

```diff
@@ -0,0 +1,57 @@
+using ExampleMod.Content.Items;
+using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using ReLogic.Content;
+using Terraria;
+using Terraria.DataStructures;
+using Terraria.ModLoader;
+
+namespace ExampleMod.Common
+{
+	public class ExamplePlayerDrawLayer : PlayerDrawLayer
+	{
+		private Asset<Texture2D> exampleItemTexture;
+
+		//Returning true in this property makes this layer appear on the minimap player head icon.
+		public override bool IsHeadLayer => true;
+
+		public override bool GetDefaultVisiblity(PlayerDrawSet drawInfo) {
+			//The layer will be visible only if the player is holding an ExampleItem in their hands. Or if another modder forces this layer to be visible.
+			return drawInfo.drawPlayer.HeldItem?.type == ModContent.ItemType<ExampleItem>();
+
+			// If you'd like to reference another PlayerDrawLayer's visibility,
+			// you can do so by getting its instance via ModContent.GetInstance<OtherDrawLayer>(), and calling GetDefaultVisiblity on it
+		}
+
+		// This layer will be a 'child' of the head layer, and draw before (beneath) it.
+		// If the Head layer is hidden, this layer will also be hidden.
+		// If the Head layer is moved, this layer will move with it.
+		public override Position GetDefaultPosition() => new Before(PlayerDrawLayers.Head);
```

Comment:
This honestly is very confusing when I Read this. The comment says it's a child of the head layer, so personally I would expect something like an .Add on the head layer. Except... is says it goes before the head layer?

Reply:
Also is there are reason this is a method and not a get only property ?

Reply:
Can make it a property.
Can you suggest better comnents?
It's a child so it inherits visibility, transform, and will move with the parent if someone reorders it.

It renders _Before_ (behind) the head layer

Reply:
Well my comment is mostly regarding how, the way it is used, it isn't apparent (to me) that it's a child of Head, only that it goes before it. How can you see it is a child, or are you just supposed to know that's 'how it works'?

Reply:
In any context, X being a child of Y shouldn't mean that it's rendered after it. It just means that it's attached to it, while still being able to have relative (local) offsets.

Reply:
Yeah, but where can I see this layer is a child of the Head layer?

Reply:
programmatically, you can see it in the head layer's `Children` list. If your suggestion is that we name this something like `BeforeParent` and `AfterParent` that's completely reasonable.